components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/HybridPolicyPersistenceManagerTest.java

```diff
@@ -0,0 +1,422 @@
+                samplePDPPolicy2.getPolicy());
+    }
+
+    @Test(priority = 25)
+    public void testGetPolicyOrderInDatabase() throws Exception {
+
+        policyPersistenceManager.addOrUpdatePolicy(samplePAPPolicy1, true);
+        policyPersistenceManager.addOrUpdatePolicy(samplePAPPolicy3, true);
+
+        policyPersistenceManager.addPolicy(samplePDPPolicy1);
+        policyPersistenceManager.addPolicy(samplePDPPolicy3);
+
+        // Verify the policy order.
+        assertEquals(policyPersistenceManager.getPolicyOrder(samplePDPPolicy1.getPolicyId()), 0);
+        assertEquals(jdbcPolicyPersistenceManager.getPolicyOrder(samplePDPPolicy1.getPolicyId()), 0);
+        assertEquals(policyPersistenceManager.getPolicyOrder(samplePDPPolicy3.getPolicyId()),
+                samplePDPPolicy3.getPolicyOrder());
+        assertEquals(jdbcPolicyPersistenceManager.getPolicyOrder(samplePDPPolicy3.getPolicyId()),
+                samplePDPPolicy3.getPolicyOrder());
+    }
+
+    @Test(priority = 26)
+    public void testGetPolicyOrderInRegistry() throws Exception {
+
+        registryPolicyPersistenceManager.addOrUpdatePolicy(samplePAPPolicy1, true);
+        registryPolicyPersistenceManager.addOrUpdatePolicy(samplePAPPolicy3, true);
+
+        registryPolicyPersistenceManager.addPolicy(samplePDPPolicy1);
+        registryPolicyPersistenceManager.addPolicy(samplePDPPolicy3);
+
+        // Verify the policy order.
+        assertEquals(policyPersistenceManager.getPolicyOrder(samplePDPPolicy1.getPolicyId()), 0);
+        assertEquals(registryPolicyPersistenceManager.getPolicyOrder(samplePDPPolicy1.getPolicyId()), 0);
+
+        assertEquals(policyPersistenceManager.getPolicyOrder(samplePDPPolicy3.getPolicyId()),
+                samplePDPPolicy3.getPolicyOrder());
+        assertEquals(registryPolicyPersistenceManager.getPolicyOrder(samplePDPPolicy3.getPolicyId()),
+                samplePDPPolicy3.getPolicyOrder());
+    }
+
+    @Test(priority = 27)
+    public void testListPDPPolicyInDatabase() throws Exception {
+
+        policyPersistenceManager.addOrUpdatePolicy(samplePAPPolicy1, true);
+        policyPersistenceManager.addOrUpdatePolicy(samplePAPPolicy2, true);
+        policyPersistenceManager.addOrUpdatePolicy(samplePAPPolicy3, true);
+
+        policyPersistenceManager.addPolicy(samplePDPPolicy1);
+        policyPersistenceManager.addPolicy(samplePDPPolicy2);
+        policyPersistenceManager.addPolicy(samplePDPPolicy3);
+
+        // Verify the number of published policies.
+        List<String> policyIds = policyPersistenceManager.listPublishedPolicyIds();
+        assertEquals(policyIds.size(), 3);
+        List<String> dbPolicyIds = jdbcPolicyPersistenceManager.listPublishedPolicyIds();
+        assertEquals(dbPolicyIds.size(), 3);
+
+        // Verify the number of ordered policy identifiers.
+        String[] orderedPolicyIdentifiers = jdbcPolicyPersistenceManager.getOrderedPolicyIdentifiers();
```

Comment:
This validation exists in registry test, but not here.. Shall we make it consistent?
```
        String[] orderedPolicyIdentifiers = policyPersistenceManager.getOrderedPolicyIdentifiers();
        assertEquals(orderedPolicyIdentifiers.length, 3);
```

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/compare/2f503927491810d4d2cc8b7ea09b65f587525e2e..585e1901da74cee150023ab7584efca46ed666d1